SpatialGDK/Source/SpatialGDK/Private/Interop/InitialOnlyFilter.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/InitialOnlyFilter.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialOSDispatcherInterface.h"
+#include "Interop/SpatialReceiver.h"
+
+DEFINE_LOG_CATEGORY(LogInitialOnlyFilter);
+
+namespace SpatialGDK
+{
+InitialOnlyFilter::InitialOnlyFilter(USpatialNetDriver* InNetDriver)
+	: NetDriver(InNetDriver)
+{
+}
+
+bool InitialOnlyFilter::HasInitialOnlyData(Worker_EntityId EntityId) const
+{
+	return (RetrievedInitialOnlyData.Find(EntityId) != nullptr);
+}
+
+bool InitialOnlyFilter::HasInitialOnlyDataOrRequest(Worker_EntityId EntityId)
+{
+	if (HasInitialOnlyData(EntityId))
+	{
+		return true;
+	}
+
+	if (InflightInitialOnlyEntities.Find(EntityId) != nullptr)
+	{
+		return false;
+	}
+
+	PendingInitialOnlyEntities.Add(EntityId);
+	return false;
+}
+
+void InitialOnlyFilter::FlushRequests()
+{
+	if (PendingInitialOnlyEntities.Num() == 0)
+	{
+		return;
+	}
+
+	TArray<Worker_Constraint> EntityConstraintArray;
+	TSet<Worker_EntityId_Key> EntitiesRequested(MoveTemp(PendingInitialOnlyEntities));
```

Comment:
What is this for? Why not use PendingInitialOnlyEntities and then clear it after moving once at the end?